AK/StringUtils.cpp

```diff
@@ -132,17 +136,21 @@ Optional<T> convert_to_uint(StringView str, TrimWhitespace trim_whitespace)
         return {};
 
     T value = 0;
-    auto const characters = string.characters_without_null_termination();
+    auto const* characters = string.characters_without_null_termination();
 
-    for (size_t i = 0; i < string.length(); i++) {
-        if (characters[i] < '0' || characters[i] > '9')
-            return {};
+    size_t i = 0;
+    while (characters[i] == '0' && i < string.length())
+        ++i;
 
-        if (__builtin_mul_overflow(value, 10, &value))
-            return {};
+    if ((string.length() - i) > NumericLimits<T>::max_digits())
+        return {};
```

Comment:
Here an example would be `19000000000000000000` which is 20 digits but larger than $2^{64}$